### PR TITLE
Add fullscreen toggle

### LINKED ISSUE #163: Add button for fullscreen

### PR DESCRIPTION
**Pull request by AndreasWintherMoen**
Closes #163 

I wasn't sure how to implement this but I think it turned out quite well. We'll probably have to test it on Mac and Windows as well to make sure there's no weird behaviour, but I've tested in Firefox and Chrome on Ubuntu and it works well here.

A lot of different components had to render differently when in fullscreen mode, so I used the Context API to share state across the whole app. I haven't used that much before, but I think it's implemented correctly and according to standards?

### CODE REVIEW

File: frontend/src/components/atoms/GraphSidebar.tsx [hunk 1]
```diff
@@ -1,17 +1,32 @@
...
+const GraphSidebar: React.FC<GraphSidebarProps> = ({ onSubgoalFilter }: GraphSidebarProps) => {
+  const { isFullscreen } = useContext(FullscreenContext);
+
+  return (
+    <Stack
+      bg="white"
+      borderRadius="lg"
+      p="5"
+      w="12vw"
+      // h={isFullscreen ? '80%' : '100%'}
```
Comment (erikskaar) [inline frontend/src/components/atoms/GraphSidebar.tsx:18]:
```suggestion
```

File: frontend/src/context/FullscreenContext.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+});
+
+export const FullscreenProvider: React.FC<ProviderProps> = ({ children }: ProviderProps) => {
+  const [isFullscreen, setIsFullscreen] = useState(false);
+  return (
+    <FullscreenContext.Provider
+      value={{
+        isFullscreen,
+        toggleFullscreen: () => {
+          console.log(`toggle fullscreen. New value: ${!isFullscreen}`);
```
Comment (erikskaar) [inline frontend/src/context/FullscreenContext.tsx:24]:
```suggestion
```

**Review by erikskaar — CHANGES_REQUESTED** [review]
I'm not too familiar with Context, but what exactly does it provide that we can't use Redux for? As far as i know, Redux can do everything Context does, but that might be wrong. Everything looks good though :)

File: frontend/src/components/atoms/Graph.tsx [hunk 3]
```diff
@@ -67,8 +70,18 @@ const Graph: React.FC<GraphProps> = ({ nodeFilter }: GraphProps) => {
...
       <svg id="svgGraph" height="100%" width="100%" ref={svgRef} />
+      <Button
+        position="absolute"
+        right="0px"
+        bottom="0px"
+        bgColor="transparent"
+        onClick={toggleFullscreen}
+        zIndex={1}
+      >
+        <AiOutlineFullscreen fontSize="32px" />
```
Comment (yvindo) [inline frontend/src/components/atoms/Graph.tsx:83]: isFullscreen ? thisButton  :  buttonWithArrowsInwards;

**Comment by benedihm** [comment]
Works fine on mac!

**Review by yvindo — COMMENTED** [review]
Looks great! Love the use of the isFullscreen boolean to toogle content returns.

File: frontend/src/components/molecules/GraphContainer.tsx [hunk 4]
```diff
@@ -20,8 +22,17 @@ const GraphContainer: React.FC = () => {
   };
 
   return (
-    <Stack bg="gray.400" h="80vh" spacing="5" p="10">
-      <SearchBar limit={5} />
+    <Stack
+      bg="gray.400"
+      h={isFullscreen ? '100vh' : '80vh'}
+      w={isFullscreen ? '100vw' : ''}
```
[the remark this answers is not in the document]
Comment (AndreasWintherMoen) [reply frontend/src/components/molecules/GraphContainer.tsx:28]: width='' is the same as not setting width at all. Typescript and ESLint don't let me provide conditional props, so I had to solve it like that.

**Comment by AndreasWintherMoen** [comment]
Fixed requested changes. Will have to change how fullscreen is rendered after #166 though, so I think we should merge that one first. We don't have to stress about the fullscreen feature before user testing on Monday.

**Comment by AndreasWintherMoen** [comment]
> I'm not too familiar with Context, but what exactly does it provide that we can't use Redux for? As far as i know, Redux can do everything Context does, but that might be wrong. Everything looks good though :)

Yea I guess we could use Redux. Context just felt simpler because it was such a small thing, but I suppose it's unnecessary to have both Redux and Context in the same app. If we don't end up merging this before user tests I may do it with Redux instead.

**Review by erikskaar — APPROVED** [review]

**Review by erikskaar — APPROVED** [review]
🇹🇩

**Pull request merged**